### PR TITLE
[ Milestone 3 ]  Associations

### PR DESCRIPTION
**Pull request by CakeInTech**
- Created a class Classroom with the following:
     - @label instance variable, should be initialized in the constructor.
     - Setter and getter for @label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student. 
- Created a class Book with the following:
     - @title and @author instance variables, should be initialized in the constructor.
     - Setters and getters for instance variables (remember about attr_accessor).
- Created a class Rental with the following:
     - @date instance variable, should be initialized in the constructor.
     - Setter and getter for @date (remember about attr_accessor).
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (tufoinnkuo10) [inline classroom.rb:4]: - [x] nice work. please kindly add setter and getter for @student
eg
`attr_reader :students`

File: student.rb [hunk 2]
```diff
@@ -9,4 +9,9 @@ def initialize(age, classroom, name = 'Unknown', parent_permission: true)
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
```
Comment (tufoinnkuo10) [inline student.rb:14]: - [x] nice job. please kindly add setter and getter for @classroom
eg
`attr_reader :classroom`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
```
Comment (tufoinnkuo10) [inline book.rb:2]: - [x] please ensure that Getters and Setters for `@rentals` should be present

File: person.rb [hunk 4]
```diff
@@ -13,9 +9,15 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (tufoinnkuo10) [inline person.rb:12]: - [x] nice work. please ensure that Getters and Setters for `@rentals` should be present

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @MohamedCK 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Git-flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: classroom.rb (hunk 1 above)
Comment (CakeInTech) [reply classroom.rb:4]: Hi @tufoinnkuo10 thanks for the reminder 😄 .

File: student.rb (hunk 2 above)
Comment (CakeInTech) [reply student.rb:14]: Hi @tufoinnkuo10 Thanks for the reminder, It's done. 😄

File: book.rb (hunk 3 above)
Comment (CakeInTech) [reply book.rb:2]: Hi @tufoinnkuo10  Thanks for the reminder, It's done. 😄

File: person.rb (hunk 4 above)
Comment (CakeInTech) [reply person.rb:12]: I did that thanks so much! 😄

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @MohamedCK 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
✔️Linters are passing✔️
✔️Professional README✔️
✔️Correct Git flow✔️
✔️Descriptive PR✔️

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by CakeInTech** [comment]
Thank you for the review @tufoinnkuo10  😄

**Pull request merged**